RebelTechnology/OpenWare
# feature/acdc

Updated or new support for AC/DC, XibecaDev, BioSignals. QSPI flash. Added CRC checksum to stored resources (not backwards compatible).

- testing SMT version
- fix ADC master gain enable
- manage parameters
- wip
- reset program index, register input/output parameters
- main menu options
- navigate patch list
- prevent overwriting data with patch name
- fix typo/missing return
- only set config mode on startup if default patch not available
- updated bootloader
- moved error messages
- triple DMA space
- configurable blocksize
- allow default blocksize override
- set vtor and disable/enable irqs
- wip
- updated reset loop and added optional jump to system DFU bootloader
- update cube, added hspi5 for external SPI FLASH
- draft dev board
- updated config
- config updates
- enabled exti
- reduced peripheral config
- screen changes
- updated usb from OwlPedal2 project
- check uploaded size
- fixed for non-audio compilation
- usb audio wip
- apparently working usb emg signal
- fixed KX122 accelerator data format
- bootloader magic address should be volatile
- updated heap zero-fill, added optional pre-fx USB conversion
- udpated codec reset
- added memory and instruction barriers before reset
- include usbh i/f
- added midi uart dma
- faster GPIOs
- updated reset procedure
- stop at third reset
- comments
- refactoring nor flash code
- refactoring spi and qspi flash
- refactoring spi/qspi flash
- qspi wip
- notes
- increasing stack/heap size
- qspi wip
- added notes
- flash qspi wip
- qspi flash wip
- updated qspi, works in mixed/memory mapped mode
- renamed source file
- factored out device reset functions to support.c
- support patch/resource listing and deletion
- refactored MidiBoot
- renamed MidiBoot projects for consistency
- MPU updates
- invalidate cache (if enabled) and clear all 8 interrupt registers
- updated quad enable and code cleanup
- red boards wip
- v22.2.2
- fixed for renamed MidiBoot projects
- disable internal flash storage for Genius
- updated sysex target
- added device_cache_invalidate() and header guard
- reverted MPU config
- blank header memory before loading data
- cache and heap management
- notes
- re-enable IWDG
- configurable dac handle
- watchdog refactor
- wait for program to exit when changing patches
- increased stack sizes
- updated MPU settings
- return copy status
- wip
- refactored resources, added checksum to ResourceHeader, reduced max resource name size
- read data in 32k blocks
- Protect tasks with critical sections. Use main stack for utility tasks.
- resource, patch, spi updates
- fully configured
- adjusted for library updates
- don't use tick/idle hooks
- common freertos.c
- tidy up code
- updated libs
- renamed project
- v22.3.0
